### ISSUE TITLE
Image training 

### ISSUE THREAD

**Issue by WeheliyeHashi** [kirkegaardlab/deeptangle#14]
![ouput](https://user-images.githubusercontent.com/88719125/227230056-b0f23fd8-0a0b-411b-833b-2bbf7e0c2f21.png)

https://user-images.githubusercontent.com/88719125/227230822-b86dbf1b-d981-49ac-b90d-743d093044a2.mp4

Hello there,

I apologize for the numerous questions and concerns I have raised earlier. I hope that this will be my final issue regarding the deeptangle code.

I have attempted to train my own data using an experiment involving crawling worms, as shown in the attached video. The input is the actual video, and the labels are represented by the red lines on top of the worms in the video.

After training the network with my data for five days using your training code, I obtained a model that produces splines that are much smaller than the actual size of the worms in my experiments, which are approximately 240 to 250 pixels in size. Is there anything that can be done to adjust the model to match the image's splines?

The following are the parameters I input into my training model:

N_suggestions =8
Latent_dim =8
Npca =36
Sigma =20
Thank you for your help!

**Comment by juliusbierk** [comment]
Hi,
It is hard for me judge what could have gone wrong without knowing more, but it definitely looks like something fundamental has gone wrong.
Would you mind sharing your data with me and the changes you have made to the code? Then I could have a look at what the problem could be?
Best wishes,
Julius

**Comment by WeheliyeHashi** [comment]
Thanks @juliusbierk. Sure! Would you like me to share the modified code and data through this email: Julius.Kirkegaard@nbi.ku.dk

**Comment by juliusbierk** [comment]
Yes, that would be fine. Super -- I'll get back to you once I've had a look.

**Comment by WeheliyeHashi** [comment]
Thanks!

**Comment by juliusbierk** [comment]
Hi @WeheliyeHashi 

Thank you for sending me your data and the updates you've made to the code.
I have a few comments for you:

1) You are storing your data very redundantly. Your hdf5 file is 250 GB. If I simply store png's and labels I can store your entire dataset in 150 MB's -- a lot easier to work with.

2) Your videos have very few worms. Our method is designed to alleviate problems that happens at much higher densities, and at much lower resolution. To make the analysis of your data work, you should be able to get away with much simpler approaches.

3) With regards to your problem, as far as I can identify the problem stems from the fact that you initialise the PCA transform with simulated worms that do not match your data. You can get around this by changing out the synthetic PCA procedure

```python
def init_pca(key, synth_generator, n_components: int):
    X = synth_generator.init_pca(key)
    X = X.reshape(len(X), -1)
    pca = PCA(n_components=n_components)
    pca.fit(X)
    A = pca.components_
    A -= A.mean(1)[:, None]
    A = jnp.array(A)
    return A
```

with something like

```python
def init_pca(X, n_components: int):
    X = X.reshape(len(X), -1)
    pca = PCA(n_components=n_components)
    pca.fit(X)
    A = pca.components_
    A -= A.mean(1)[:, None]
    A = jnp.array(A)
    return A
```

where `X` is then simply a `[N, 49, 2]` matrix that contains all your training labels.

I hope this makes sense?

Best wishes,
Julius

**Comment by WeheliyeHashi** [comment]
![image](https://user-images.githubusercontent.com/88719125/228194788-58a4a9e4-18c2-4384-8ec3-aaea3dae6583.png)

https://user-images.githubusercontent.com/88719125/228195378-07ccbbaa-7700-449f-a93c-a959ac4323fd.mov


https://user-images.githubusercontent.com/88719125/228196820-2e3935d0-bed2-43ff-a0e0-4cfc616c7af5.mp4

**Comment by WeheliyeHashi** [comment]
@juliusbierk ..just to confirm point 3 makes sense ...

**Comment by WeheliyeHashi** [comment]
Hi @juliusbierk , you can close this issue. I have managed to improve my model..see an snap shot for crawling worms. Thanks again for your help. 
![val_dation1](https://github.com/kirkegaardlab/deeptangle/assets/88719125/486358af-dd67-4b5a-8e32-4d352112535b)
